<a href="https://colab.research.google.com/github/nitish6121999/US-HOME-PRICE-PREDICTION/blob/main/Home_LLC__Data_Preparation_Assignmnet.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name**    - U.S HOME PRICE PREDICTION



##### **Project Type**    - Supervised Machine Learning Regression
##### **Created by**      - Nitish N Naik

# **GitHub Link -**

https://github.com/nitish6121999/US-HOME-PRICE-PREDICTION

# **Problem Statement**



The aim of this study is to develop a robust predictive model for US home prices leveraging comprehensive datasets encompassing various economic, societal, and market-related factors. The objective is to accurately predict future home prices based on historical trends and the interplay of influential variables.

Steps Involved:

Data Collection: Gather data from credible online sources encompassing US home pricing, real estate metrics, and a comprehensive set of factors known to influence housing markets.

Dataset Merging and Understanding: Consolidate and merge datasets to create a unified dataset. Understand the structure, contents, and relationships within the data.

Focus on US Home Pricing and Real Estate: Delve deep into the dynamics of US home pricing and real estate markets to comprehend patterns, trends, and influencing factors.

Factor Identification and Understanding: Identify, analyze, and understand the multitude of factors affecting home prices as potential independent variables.

Exploratory Data Analysis (EDA): Perform exploratory data analysis to uncover correlations, trends, outliers, and patterns within the dataset. This step involves data cleaning, visualization, and statistical analysis.

Model Development: Utilize machine learning or statistical modeling techniques to build a predictive model. Train the model using historical data and the identified factors to predict future home prices accurately.

Model Evaluation and Validation: Validate the model's accuracy, robustness, and predictive capabilities using suitable metrics and validation techniques.

Iterative Refinement: Continuously refine the model based on feedback from validation results. Fine-tune parameters or consider additional factors to enhance predictive accuracy.*



### Factors considered

House price Index 2001-2023

Average Sales Price of Houses Sold for the United States

Consumer Price Index for All Urban Consumers Housing in U.S. City

Average Economic Policy Uncertainty Index for United States (USEPUINDXD)

Government subsidies Federal Housing L312051A027NBEA

Homeownership Rate in the United States RHORUSQ156N

Interest Rates and Price Indexes 2001-2023.csv

Monthly Supply of New Houses in the United States

Mortgage Average in the United States MORTGAGE30US

Net housing value added Subsidies

Public Transportation in U.S. City Average

Real Median Household Income in the United States 2001-2022

Unemployeement rate 2002-2023

Working Age Population Aged 15-64

# ***Let's Begin !***

## ***1. Know Your Data***

### Import Libraries

In [39]:
# Import Libraries
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

### Dataset Loading

In [40]:
# Load Dataset

In [41]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


### Dataset First View

In [42]:
# Dataset First Look
#path= '/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/House price Index 2001-2023.csv'

df_hpi=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/House price Index 2001-2023.csv',names=["DATE","Price-index"], skiprows = 1)
#df_cons_price_index = pd.read_csv("Construction_Price.csv", names = ["DATE", "Cons_Materials"], skiprows = 1)

In [43]:
#Changing dtype of date column
df_hpi["DATE"] = pd.to_datetime(df_hpi["DATE"])

#Selecting data post JUNE 2001
mask = df_hpi["DATE"] <= "2022-12-01"
df_hpi = df_hpi[mask]

#Resetting Index
df_hpi.reset_index(inplace = True)
df_hpi.drop(columns = ["index"], inplace = True)

# Creating "Year" and "Month" columns
df_hpi["Year"] = pd.DatetimeIndex(df_hpi["DATE"]).year
df_hpi["Month"] = pd.DatetimeIndex(df_hpi["DATE"]).month
print(df_hpi.shape)
df_hpi.head()

(252, 4)


DATE  Price-index  Year  Month
0 2002-01-01      117.144  2002      1
1 2002-01-02      117.844  2002      1
2 2002-01-03      118.687  2002      1
3 2002-01-04      119.610  2002      1
4 2002-01-05      120.724  2002      1

### Dataset Rows & Columns count

In [44]:
# Dataset Rows & Columns count
df_hpi.shape

(252, 4)

### Dataset Information

###Unemployeement rate

In [45]:
# importing Dataset from different sources
df_unem=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Unemployeement rate 2002-2023.csv', encoding='unicode_escape', parse_dates=[0])
print(df_unem.shape)

df_unem.head()

(262, 2)


DATE  UNRATE
0 2002-01-01     5.7
1 2002-02-01     5.7
2 2002-03-01     5.7
3 2002-04-01     5.9
4 2002-05-01     5.8

In [46]:
 #df_hpi = pd.concat([df_hpi,df_unem], axis = 1)
# df_hpi.head()



Average Sales Price of Houses Sold for the United States


In [47]:
df_hs=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Average Sales Price of Houses Sold for the United States ASPUS.csv',  names = ["DATE", "avg_sales_price"], skiprows = 1)
print(df_hs.shape)

df_hs.head()

(91, 2)


DATE  avg_sales_price
0  2001-01-01         211000.0
1  2001-04-01         211200.0
2  2001-07-01         207800.0
3  2001-10-01         214200.0
4  2002-01-01         227600.0


Consumer Price Index for All Urban Consumers Housing in U.S. City



In [48]:
df_cpi=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Consumer Price Index for All Urban Consumers Housing in U.S. City Average 2001-2023.csv', names = ["DATE", "con_price_index"], skiprows = 1)
print(df_cpi.shape)

df_cpi.head()

(275, 2)


DATE  con_price_index
0  2001-01-01            174.1
1  2001-02-01            174.7
2  2001-03-01            175.4
3  2001-04-01            175.4
4  2001-05-01            175.9

Average Economic Policy Uncertainty Index for United States (USEPUINDXD)



In [49]:
df_ep=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Economic Policy Uncertainty Index for United States (USEPUINDXD).csv', names = ["DATE", "avg_Eco_policy"], skiprows = 1)
print(df_ep.shape)

df_ep.head()

(276, 2)


DATE  avg_Eco_policy
0  2001-01-01      123.213871
1  2001-02-01      122.261071
2  2001-03-01      124.462258
3  2001-04-01      105.174333
4  2001-05-01       96.311935

Government subsidies Federal Housing L312051A027NBEA



In [50]:
df_gs=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Government subsidies Federal Housing L312051A027NBEA.csv', names = ["DATE", "govt_subsidies"], skiprows = 1)
print(df_gs.shape)

df_gs.head()

(22, 2)


DATE  govt_subsidies
0  2001-01-01          20.573
1  2002-01-01          24.183
2  2003-01-01          25.930
3  2004-01-01          27.201
4  2005-01-01          27.651

Homeownership Rate in the United States RHORUSQ156N



In [51]:
df_ho=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Homeownership Rate in the United States RHORUSQ156N.csv', names = ["DATE", "homeownership_rate"], skiprows = 1)
print(df_ho.shape)

df_ho.head()

(91, 2)


DATE  homeownership_rate
0  2001-01-01                67.5
1  2001-04-01                67.7
2  2001-07-01                68.1
3  2001-10-01                68.0
4  2002-01-01                67.8

Interest Rates and Price Indexes 2001-2023.csv



In [52]:
df_ir=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Interest Rates and Price Indexes 2001-2023.csv', names = ["DATE", "Interest_rate"], skiprows = 1)
print(df_ir.shape)

df_ir.head()

(87, 2)


DATE  Interest_rate
0  01-01-2002           6972
1  01-04-2002           6813
2  01-07-2002           6289
3  01-10-2002           6075
4  01-01-2003           5841

Monthly Supply of New Houses in the United States


In [53]:
df_ms=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Monthly Supply of New Houses in the United States.csv', names = ["DATE", "monthly_new_house"], skiprows = 1)
print(df_ms.shape)

df_ms.head()

(275, 2)


DATE  monthly_new_house
0  2001-01-01                3.8
1  2001-02-01                3.7
2  2001-03-01                3.8
3  2001-04-01                3.9
4  2001-05-01                4.0


Mortgage Average in the United States MORTGAGE30US


In [54]:
df_ma=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/mortgage_rate.csv', names = ["DATE", "mortgage"], skiprows = 1)
print(df_ma.shape)

df_ma.head()

(22, 2)


DATE  mortgage
0  2002      6.54
1  2003      5.83
2  2004      5.84
3  2005      5.87
4  2006      6.41


Net housing value added Subsidies


In [55]:
df_nhv=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Net housing value added Subsidies.csv', names = ["DATE", "net_housing_value"], skiprows = 1)
print(df_nhv.shape)

df_nhv.head()

(22, 2)


DATE  net_housing_value
0  2001-01-01             21.094
1  2002-01-01             24.429
2  2003-01-01             26.503
3  2004-01-01             28.585
4  2005-01-01             31.688


Public Transportation in U.S. City Average


In [56]:
df_pt=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Public Transportation in U.S. City Average.csv', names = ["DATE", "transportation"], skiprows = 1)
print(df_pt.shape)

df_pt.head()

(275, 2)


DATE  transportation
0  2001-01-01           210.2
1  2001-02-01           212.1
2  2001-03-01           210.0
3  2001-04-01           208.3
4  2001-05-01           209.3


Real Median Household Income in the United States 2001-2022



In [57]:
df_hi=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Real Median Household Income in the United States 2001-2022.csv', names = ["DATE", "household_income"], skiprows = 1)
print(df_hi.shape)

df_hi.head()

(22, 2)


DATE  household_income
0  2001-01-01             66360
1  2002-01-01             65820
2  2003-01-01             65860
3  2004-01-01             65760
4  2005-01-01             66780

Working Age Population Aged 15-64

In [58]:
df_wp=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Working Age Population Aged 15-64.csv', names = ["DATE", "population"], skiprows = 1)

print(df_wp.shape)

df_wp.head()

(274, 2)


DATE    population
0  2001-01-01  1.804166e+08
1  2001-02-01  1.805777e+08
2  2001-03-01  1.806361e+08
3  2001-04-01  1.809055e+08
4  2001-05-01  1.810146e+08

Fedral funds effective rate

In [59]:
df_ff=pd.read_csv('/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/Federal Funds Effective Rate FEDFUNDS.csv',  names = ["DATE", "fedral_rate"], skiprows = 1)
print(df_ff.shape)

df_ff.head()

(264, 2)


DATE  fedral_rate
0  2001-01-01         5.98
1  2001-02-01         5.49
2  2001-03-01         5.31
3  2001-04-01         4.80
4  2001-05-01         4.21

GDP

In [60]:
df_gdp=pd.read_csv("/content/drive/MyDrive/Access file/Extras/HOME LLC/Dataset/GDP.csv", names = ["DATE", "Per_Capita_GDP"], skiprows = 1)
print(df_gdp.shape)

df_gdp.head()

(91, 2)


DATE  Per_Capita_GDP
0  2001-01-01       10470.231
1  2001-04-01       10599.000
2  2001-07-01       10598.020
3  2001-10-01       10660.465
4  2002-01-01       10783.500

df_bymonth = [df_hpi, df_unem, df_cpi, df_ep,df_ms,df_pt,df_wp,df_ff]

In [61]:
df_bymonth = [df_ep,df_unem,df_cpi,df_ms,df_pt,df_wp,df_ff]
for df1 in df_bymonth:
  df_hpi = pd.concat([df_hpi,df1],axis=1)
print(df_hpi.shape)
df_hpi.head()

(276, 18)


DATE  Price-index    Year  Month        DATE  avg_Eco_policy  \
0 2002-01-01      117.144  2002.0    1.0  2001-01-01      123.213871   
1 2002-01-02      117.844  2002.0    1.0  2001-02-01      122.261071   
2 2002-01-03      118.687  2002.0    1.0  2001-03-01      124.462258   
3 2002-01-04      119.610  2002.0    1.0  2001-04-01      105.174333   
4 2002-01-05      120.724  2002.0    1.0  2001-05-01       96.311935   

        DATE  UNRATE        DATE  con_price_index        DATE  \
0 2002-01-01     5.7  2001-01-01            174.1  2001-01-01   
1 2002-02-01     5.7  2001-02-01            174.7  2001-02-01   
2 2002-03-01     5.7  2001-03-01            175.4  2001-03-01   
3 2002-04-01     5.9  2001-04-01            175.4  2001-04-01   
4 2002-05-01     5.8  2001-05-01            175.9  2001-05-01   

   monthly_new_house        DATE  transportation        DATE    population  \
0                3.8  2001-01-01           210.2  2001-01-01  1.804166e+08   
1                3.7  2001-02-01           212.1  2001-02-01  1.805777e+08   
2                3.8  2001-03-01           210.0  2001-03-01  1.806361e+08   
3                3.9  2001-04-01           208.3  2001-04-01  1.809055e+08   
4                4.0  2001-05-01           209.3  2001-05-01  1.810146e+08   

         DATE  fedral_rate  
0  2001-01-01         5.98  
1  2001-02-01         5.49  
2  2001-03-01         5.31  
3  2001-04-01         4.80  
4  2001-05-01         4.21

In [62]:
df_hpi.isnull().sum()

DATE                 24
Price-index          24
Year                 24
Month                24
DATE                  0
avg_Eco_policy        0
DATE                 14
UNRATE               14
DATE                  1
con_price_index       1
DATE                  1
monthly_new_house     1
DATE                  1
transportation        1
DATE                  2
population            2
DATE                 12
fedral_rate          12
dtype: int64

In [63]:
df_hpi.dropna(inplace=True)

In [64]:
df_hpi.head()

DATE  Price-index    Year  Month        DATE  avg_Eco_policy  \
0 2002-01-01      117.144  2002.0    1.0  2001-01-01      123.213871   
1 2002-01-02      117.844  2002.0    1.0  2001-02-01      122.261071   
2 2002-01-03      118.687  2002.0    1.0  2001-03-01      124.462258   
3 2002-01-04      119.610  2002.0    1.0  2001-04-01      105.174333   
4 2002-01-05      120.724  2002.0    1.0  2001-05-01       96.311935   

        DATE  UNRATE        DATE  con_price_index        DATE  \
0 2002-01-01     5.7  2001-01-01            174.1  2001-01-01   
1 2002-02-01     5.7  2001-02-01            174.7  2001-02-01   
2 2002-03-01     5.7  2001-03-01            175.4  2001-03-01   
3 2002-04-01     5.9  2001-04-01            175.4  2001-04-01   
4 2002-05-01     5.8  2001-05-01            175.9  2001-05-01   

   monthly_new_house        DATE  transportation        DATE    population  \
0                3.8  2001-01-01           210.2  2001-01-01  1.804166e+08   
1                3.7  2001-02-01           212.1  2001-02-01  1.805777e+08   
2                3.8  2001-03-01           210.0  2001-03-01  1.806361e+08   
3                3.9  2001-04-01           208.3  2001-04-01  1.809055e+08   
4                4.0  2001-05-01           209.3  2001-05-01  1.810146e+08   

         DATE  fedral_rate  
0  2001-01-01         5.98  
1  2001-02-01         5.49  
2  2001-03-01         5.31  
3  2001-04-01         4.80  
4  2001-05-01         4.21

In [65]:
date_columns = [col for col in df_hpi.columns if col == 'DATE']
columns_to_drop = date_columns[1:]

# Drop columns except the first occurrence of 'DATE'
df_hpi.drop(columns=columns_to_drop, inplace=True)

In [66]:
df_hpi.shape

(252, 10)

In [67]:
df_hpi.head()

Price-index    Year  Month  avg_Eco_policy  UNRATE  con_price_index  \
0      117.144  2002.0    1.0      123.213871     5.7            174.1   
1      117.844  2002.0    1.0      122.261071     5.7            174.7   
2      118.687  2002.0    1.0      124.462258     5.7            175.4   
3      119.610  2002.0    1.0      105.174333     5.9            175.4   
4      120.724  2002.0    1.0       96.311935     5.8            175.9   

   monthly_new_house  transportation    population  fedral_rate  
0                3.8           210.2  1.804166e+08         5.98  
1                3.7           212.1  1.805777e+08         5.49  
2                3.8           210.0  1.806361e+08         5.31  
3                3.9           208.3  1.809055e+08         4.80  
4                4.0           209.3  1.810146e+08         4.21

In [68]:
df_bymonth = [df_gs,df_ma,df_nhv,df_hi,df_hs,df_ho,df_gdp,df_ir]
for df1 in df_bymonth:
  df_hpi = pd.concat([df_hpi,df1],axis=1)
print(df_hpi.shape)
df_hpi.head()

(252, 26)


Price-index    Year  Month  avg_Eco_policy  UNRATE  con_price_index  \
0      117.144  2002.0    1.0      123.213871     5.7            174.1   
1      117.844  2002.0    1.0      122.261071     5.7            174.7   
2      118.687  2002.0    1.0      124.462258     5.7            175.4   
3      119.610  2002.0    1.0      105.174333     5.9            175.4   
4      120.724  2002.0    1.0       96.311935     5.8            175.9   

   monthly_new_house  transportation    population  fedral_rate  ...  \
0                3.8           210.2  1.804166e+08         5.98  ...   
1                3.7           212.1  1.805777e+08         5.49  ...   
2                3.8           210.0  1.806361e+08         5.31  ...   
3                3.9           208.3  1.809055e+08         4.80  ...   
4                4.0           209.3  1.810146e+08         4.21  ...   

         DATE  household_income        DATE  avg_sales_price        DATE  \
0  2001-01-01           66360.0  2001-01-01         211000.0  2001-01-01   
1  2002-01-01           65820.0  2001-04-01         211200.0  2001-04-01   
2  2003-01-01           65860.0  2001-07-01         207800.0  2001-07-01   
3  2004-01-01           65760.0  2001-10-01         214200.0  2001-10-01   
4  2005-01-01           66780.0  2002-01-01         227600.0  2002-01-01   

   homeownership_rate        DATE  Per_Capita_GDP        DATE  Interest_rate  
0                67.5  2001-01-01       10470.231  01-01-2002         6972.0  
1                67.7  2001-04-01       10599.000  01-04-2002         6813.0  
2                68.1  2001-07-01       10598.020  01-07-2002         6289.0  
3                68.0  2001-10-01       10660.465  01-10-2002         6075.0  
4                67.8  2002-01-01       10783.500  01-01-2003         5841.0  

[5 rows x 26 columns]

In [69]:
df_hpi.isnull().sum()

Price-index             0
Year                    0
Month                   0
avg_Eco_policy          0
UNRATE                  0
con_price_index         0
monthly_new_house       0
transportation          0
population              0
fedral_rate             0
DATE                  230
govt_subsidies        230
DATE                  230
mortgage              230
DATE                  230
net_housing_value     230
DATE                  230
household_income      230
DATE                  161
avg_sales_price       161
DATE                  161
homeownership_rate    161
DATE                  161
Per_Capita_GDP        161
DATE                  165
Interest_rate         165
dtype: int64

In [70]:
date_columns = [col for col in df_hpi.columns if col == 'DATE']
columns_to_drop = date_columns[1:]

# Drop columns except the first occurrence of 'DATE'
df_hpi.drop(columns=columns_to_drop, inplace=True)

In [71]:
df_hpi.isnull().sum()

Price-index             0
Year                    0
Month                   0
avg_Eco_policy          0
UNRATE                  0
con_price_index         0
monthly_new_house       0
transportation          0
population              0
fedral_rate             0
govt_subsidies        230
mortgage              230
net_housing_value     230
household_income      230
avg_sales_price       161
homeownership_rate    161
Per_Capita_GDP        161
Interest_rate         165
dtype: int64

###Yearly Data Present
Govt_subsidies        230

mortgage              230

net_housing_value     230

household_income      230

###Quarterly Data present

avg_sales_price       161

homeownership_rate    161

Per_Capita_GDP        161

Interest_rate         165

In [72]:
columns_to_interpolate = ['govt_subsidies', 'mortgage', 'net_housing_value', 'household_income',
                          'avg_sales_price', 'homeownership_rate', 'Per_Capita_GDP', 'Interest_rate']

# Interpolate missing values in specified columns
df_hpi[columns_to_interpolate] = df_hpi[columns_to_interpolate].interpolate()


In [73]:
df_hpi.shape

(252, 18)

In [74]:
df_hpi.isnull().sum()

Price-index           0
Year                  0
Month                 0
avg_Eco_policy        0
UNRATE                0
con_price_index       0
monthly_new_house     0
transportation        0
population            0
fedral_rate           0
govt_subsidies        0
mortgage              0
net_housing_value     0
household_income      0
avg_sales_price       0
homeownership_rate    0
Per_Capita_GDP        0
Interest_rate         0
dtype: int64

In [75]:
df_hpi.to_csv("merged_dataset.csv")

In [76]:
df_hpi.head(10)

Price-index    Year  Month  avg_Eco_policy  UNRATE  con_price_index  \
0      117.144  2002.0    1.0      123.213871     5.7            174.1   
1      117.844  2002.0    1.0      122.261071     5.7            174.7   
2      118.687  2002.0    1.0      124.462258     5.7            175.4   
3      119.610  2002.0    1.0      105.174333     5.9            175.4   
4      120.724  2002.0    1.0       96.311935     5.8            175.9   
5      121.813  2002.0    1.0       70.574333     5.8            177.3   
6      122.888  2002.0    1.0       91.896452     5.8            177.6   
7      123.830  2002.0    1.0       80.837419     5.7            178.0   
8      124.780  2002.0    1.0      234.574333     5.7            177.4   
9      125.734  2002.0    1.0      236.406129     5.7            176.7   

   monthly_new_house  transportation    population  fedral_rate  \
0                3.8           210.2  1.804166e+08         5.98   
1                3.7           212.1  1.805777e+08         5.49   
2                3.8           210.0  1.806361e+08         5.31   
3                3.9           208.3  1.809055e+08         4.80   
4                4.0           209.3  1.810146e+08         4.21   
5                4.2           216.3  1.813565e+08         3.97   
6                4.2           216.1  1.816143e+08         3.77   
7                4.4           213.7  1.818285e+08         3.65   
8                4.4           212.7  1.820565e+08         3.07   
9                4.3           209.1  1.822366e+08         2.49   

   govt_subsidies  mortgage  net_housing_value  household_income  \
0          20.573      6.54             21.094           66360.0   
1          24.183      5.83             24.429           65820.0   
2          25.930      5.84             26.503           65860.0   
3          27.201      5.87             28.585           65760.0   
4          27.651      6.41             31.688           66780.0   
5          28.604      6.34             30.683           67520.0   
6          29.512      6.03             37.448           68610.0   
7          29.876      5.04             33.461           66280.0   
8          32.883      4.69             34.860           65850.0   
9          32.669      4.45             34.177           64300.0   

   avg_sales_price  homeownership_rate  Per_Capita_GDP  Interest_rate  
0         211000.0                67.5       10470.231         6972.0  
1         211200.0                67.7       10599.000         6813.0  
2         207800.0                68.1       10598.020         6289.0  
3         214200.0                68.0       10660.465         6075.0  
4         227600.0                67.8       10783.500         5841.0  
5         227600.0                67.6       10887.460         5507.0  
6         219100.0                68.0       10984.040         6034.0  
7         232500.0                68.3       11061.433         5919.0  
8         233100.0                68.0       11174.129         5592.0  
9         241000.0                68.0       11312.766         6159.0

EXPLORATORY DATA ANALYIS IS DONE IN THE FULL ASSIGNMENT FILE